module/VuFind/src/VuFind/RecordTab/HoldingsWorldCat2.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Holdings (WorldCat2) tab
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  RecordTabs
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_tabs Wiki
+ */
+
+namespace VuFind\RecordTab;
+
+use VuFindSearch\Backend\WorldCat2\Command\GetHoldingsCommand;
+use VuFindSearch\ParamBag;
+use VuFindSearch\Service;
+
+/**
+ * Holdings (WorldCat2) tab
+ *
+ * @category VuFind
+ * @package  RecordTabs
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_tabs Wiki
+ */
+class HoldingsWorldCat2 extends AbstractBase
+{
+    /**
+     * Constructor
+     *
+     * @param Service $searchService Search service
+     * @param array   $defaults      Default parameters to include in API requests
+     */
+    public function __construct(protected Service $searchService, protected array $defaults = [])
+    {
+    }
+
+    /**
+     * Get the on-screen description for this tab.
+     *
+     * @return string
+     */
+    public function getDescription()
+    {
+        return 'Holdings';
+    }
+
+    /**
+     * Get holdings information from WorldCat (null if none available).
+     *
+     * @return ?array
+     */
+    public function getHoldings()
```

Comment:
This means libraries holding the item... I could rename this method to getHoldingInstitutionData() if you think that would be more clear. Here's an example of what it looks like for me:

![image](https://github.com/user-attachments/assets/a6365e80-a2a2-440f-b307-46da369ec9ae)

Last I checked (which was a few weeks ago), all of the geographic functionality in the API was broken, so I wasn't able to use it. So right now it's just going to be an arbitrary list of holding institutions, without much useful information.

If you're getting an empty tab for everything, I wonder if perhaps some extra permission is needed on your API key to activate the functionality. I could ask @librarywebchic if that would help!

Reply:
@maccabeelevine if you send me the key you're using I can look in the log files to see what's going on with the holdings request.

Reply:
In debug, I don't see this HoldingsWorldCat2 class ever being called.  The path I see is that holdingsils.phtml calls the record driver's `getRealTimeHoldings()`.  WorldCat2 record driver doesn't define that, so it falls back to the DefaultRecord implementation which returns an empty result.  ILSAwareTrait would do the right thing (maybe?) but the WorldCat2 record driver doesn't use it.

I did email Karen with my API key in case that makes the difference somehow, but seems not.

Reply:
If you're looking at a WorldCat2 record, you should never see holdingsils.phtml at all. Is it possible you somehow have an outdated RecordTabs.ini in your test instance that's preventing the appropriate WorldCat2-specific tabs from being loaded?

Reply:
<sigh> yes.  I have one in `local` that was overriding your upstream change.  Tab looks good!  Having not seen this functionality in the original WorldCat driver, I assume it's the same -- it would be useful to linkify those holdings to WorldCat so the patron can do something about them, but assuming this is consistent with the old WorldCat API implementation, no reason to iterate on that right now.

Also FWIW I tried adding `heldInState = "US-HI"` to the .ini and it works great, I can see all the places I could borrow the book if I were lucky enough to be near a Hawaiian beach instead of in my office.

Reply:
The original WorldCat holdings tab had more functionality than the current version, because the original API had more functionality. There's some uncertainty about whether a separate API (possibly for-fee) is needed to get some of the data going forward. @librarywebchic is investigating, and we can enhance the tab in a separate future PR once progress is made.

Glad to hear that the limit option is working for you!